src/core/src/runtime/tensor_util.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/runtime/tensor_util.hpp"
+
+#include <filesystem>
+#include <fstream>
+
+#include "openvino/runtime/aligned_buffer.hpp"
+#include "openvino/runtime/shared_buffer.hpp"
+#include "openvino/util/mmap_object.hpp"
+
+namespace ov {
+
+namespace {
+
+struct StaticBufferAllocator {
+    std::shared_ptr<ov::AlignedBuffer> buffer;
+    bool allocated = false;  // if buffer was returned as allocated region
+
+    StaticBufferAllocator(std::shared_ptr<ov::AlignedBuffer> _buffer) : buffer(_buffer) {}
+
+    void* allocate(const size_t bytes, const size_t alignment) {
+        // TODO: Add check for alignment
+        OPENVINO_ASSERT(!allocated);
+        OPENVINO_ASSERT(bytes == buffer->size());
+        allocated = true;
+        return buffer->get_ptr();
+    }
+
+    void deallocate(void* handle, const size_t bytes, const size_t alignment) {
+        OPENVINO_ASSERT(handle == buffer->get_ptr());
+        OPENVINO_ASSERT(bytes == buffer->size());
+    }
```

Comment:
Deallocation shouldn't throw, it may be called from dtor.

Reply:
removed these checks